helm/ffc-tcg-portal/templates/config-map.yaml

```diff
@@ -15,4 +15,12 @@ data:
   DEFRA_ID_REDIRECT_URL: https://{{ .Values.ingress.endpoint }}.{{ .Values.ingress.server }}/{{ .Values.container.defraIdRedirectUrl }}
   {{- end }}
   ENVIRONMENT_CODE: {{.Values.environmentCode}}
+  {{- if (ne (.Values.useAbacoStub))}}
+  {{- if and (.Values.environmentCode) (ne (.Values.environmentCode | toString ) "snd") }}
+  ABACO_ENDPOINT: {{ .Values.container.abacoEndpoint }}-{{ .Values.environmentCode }}.{{ .Values.ingress.server }}
+  {{ else }}
+  ABACO_ENDPOINT: {{ .Values.container.abacoEndpoint }}.{{ .Values.ingress.server }}
+  {{- end }}
+  {{- end }}
```

Comment:
Needs to resolve a value if `useAbacoStub` is true.

Which would need similar logic.

eg if snd and environment code then `{{ .Values.container.stubEndpoint }}.{{ .Values.workstream }}-{{ .Values.environmentCode }}`

if not snd or environment code then `{{ .Values.container.stubEndpoint }}.{{ .Values.workstream }}`

Reply:
`stubEndpoint` you can set to `ffc-tcg-abaco-agri-stub`

Reply:
Bit confused by this bit. Is this for if the service in one of the environments is using the stub?

Does it need a new if statement for `useAbacoStub` is true or can this logic be put in the else of the first statement?

Reply:
What is the `Values.workstream` value?

Reply:
We've essentially got four possible endpoints depending on whether:

1 - we are deploying to sandpit
2 - we want to use the stub instead of the abaco endpoint

Let's say the abaco endpoint is `https://abaco.com` and the stub endpoint is `http://ffc-tcg-abaco-agri-stub`

If we want to use the abaco endpoint then in sandpit value to be `https://abaco.com` if not in sandpit we want `https://abaco-dev.com` if deploying to dev, `https://abaco-test.com` if test etc.

If we want to use the stub endpoint then in sandpit we want `http://ffc-tcg-abaco-agri-stub.ffc-tcg`  if not we want `http://ffc-tcg-abaco-agri-stub.ffc-tcg-dev` if deploying to dev, `http://ffc-tcg-abaco-agri-stub.ffc-tcg-test` if test etc

The key difference is that if we're using Abaco, then we call an https endpoint exposed via App Gateway.  If we're using the stub, we use an http request direct through the cluster nextwork